python/sparkdl/graph/builder.py

```diff
@@ -0,0 +1,268 @@
+        self.graph = graph or tf.Graph()
+        self.sess = tf.Session(graph=self.graph)
+        if keras:
+            self.keras_prev_sess = K.get_session()
+        else:
+            self.keras_prev_sess = None
+
+    def __enter__(self):
+        self.sess.as_default()
+        self.sess.__enter__()
+        if self.keras_prev_sess is not None:
+            K.set_session(self.sess)
+        return self
+
+    def __exit__(self, *args):
+        if self.keras_prev_sess is not None:
+            K.set_session(self.keras_prev_sess)
+        self.sess.__exit__(*args)
+
+    def run(self, *args, **kargs):
+        return self.sess.run(*args, **kargs)
+
+    def asGraphFunction(self, inputs, outputs, strip_and_freeze=True):
+        """
+        Export the graph in this session as a GraphFunction object
+
+        :param inputs: list, graph elements representing the inputs
+        :param outputs: list, graph elements representing the outputs
+        :param strip_and_freeze: bool, should we remove unused part of the graph and freee its values
+        """
+        if strip_and_freeze:
+            gdef = tfx.strip_and_freeze_until(outputs, self.graph, self.sess)
+        else:
+            gdef = self.graph.as_graph_def(add_shapes=True)
+        return GraphFunction(graph_def=gdef,
+                             input_names=[tfx.validated_input(self.graph, elem) for elem in inputs],
+                             output_names=[tfx.validated_output(self.graph, elem) for elem in outputs])
+
+    def importGraphFunction(self, gfn, input_map=None, name="GFN-IMPORT", **gdef_kargs):
+        """
+        Import a GraphFunction object into the current session
+
+        :param gfn: GraphFunction, an object representing a TensorFlow graph and its inputs and outputs
+        :param input_map: dict, mapping from input names to existing graph elements
+        :param name: str, the scope for all the variables in the GraphFunction's elements
+        :param gdef_kargs: other keyword elements for TensorFlow's `import_graph_def`
+        """
+        try:
+            del gdef_kargs["return_elements"]
+        except KeyError:
+            pass
+        if input_map is not None:
+            assert set(input_map.keys()) <= set(gfn.input_names), \
+                "cannot locate provided input elements in the graph"
+
+        input_names = gfn.input_names
+        output_names = gfn.output_names
+        if name is not None:
+            name = name.strip()
+            if len(name) > 0:
+                output_names = [
+                    name + '/' + op_name for op_name in gfn.output_names]
+                input_names = [
+                    name + '/' + op_name for op_name in gfn.input_names]
+
+        # When importing, provide the original output op names
+        tf.import_graph_def(gfn.graph_def,
+                            input_map=input_map,
+                            return_elements=gfn.output_names,
+                            name=name,
```

Comment:
Per https://www.tensorflow.org/api_docs/python/tf/import_graph_def, passing `None` is the default of `tf.import_graph_def` and it will add the prefix "import" to the graph.